reflex/components/component.py

```diff
@@ -668,47 +668,63 @@ def _validate_component_children(self, children: List[Component]):
             children: The children of the component.
 
         """
-        skip_parentable = all(child._valid_parents == [] for child in children)
-        if not self._invalid_children and not self._valid_children and skip_parentable:
+        no_valid_parents_defined = all(child._valid_parents == [] for child in children)
+        if (
+            not self._invalid_children
+            and not self._valid_children
+            and no_valid_parents_defined
+        ):
             return
 
         comp_name = type(self).__name__
+        allowed_components = ["Fragment", "Foreach", "Cond", "Match"]
```

Comment:
I think it would be better to actually import these components, and then do issubclass checks on those, rather than checking the name. Also, perhaps we could move the validate_child method to the sub components, so we just do `child.validate_children()` and we don't have to have all the if cases here (we can do this in the future)

Reply:
Good point

Reply:
Let's do this in a follow-up - we can set the default values to `None` within cond.